### DISCUSSION TITLE
Memory Usage

### DISCUSSION THREAD

**Discussion by abillits** [developmentseed/titiler#465] [General]
Hi,

We're using the docker image provided and we've run into an issue where Titiler increases memory consumption with every request. The amount of memory used never decreases. It'll use the entire 128GB available on the server unless we restart the container.

The logs don't contain anything other than http 200 messages and everything technically works until the server runs out of available memory.

Any idea what could be causing this?

![memory](https://user-images.githubusercontent.com/2138178/167316242-a5dbfb7c-a581-41d5-a1f4-98f2407c7727.png)
![settings](https://user-images.githubusercontent.com/2138178/167316243-d349982c-7c59-45a5-861c-9ae02d3d00aa.png)

**Comment by vincentsarago** [comment]
@idtstudios I've seen this too but I've never been able to really track down where it was. 

https://github.com/tiangolo/fastapi/issues/1624 might be an interesting issue. I was also going to try using https://github.com/bloomberg/memray to track down this but didn't had time yet.

**Comment by abillits** [comment]
Well, at least I'm not crazy. I figured I had something misconfigured but nothing I tweaked helped.

I'll see what I can find.

**Comment by abillits** [comment]
Well, this is definitely an upstream issue and not a Titiler issue. It apparently plagues numerous projects built on fastapi and/or uvicorn.

I've run out of time tonight to dig into this but I've slapped a bandage on the problem by using max_requests to restart workers. Not ideal but it'll keep things running for now.
